src/pybind/mgr/cephadm/module.py

```diff
@@ -1311,6 +1307,18 @@ def _check_valid_addr(self, host: str, addr: str) -> str:
 > ssh -F ssh_config -i ~/cephadm_private_key {self.ssh_user}@{addr}'''
             raise OrchestratorError(msg)
 
+        if ipaddress.ip_address(ip_addr).is_loopback and host == addr:
+            # if this is a re-add, use old address. otherwise error
+            try:
+                old_ip = self.inventory.get_addr(host)
+                assert old_ip != host
+            except Exception:
+                raise OrchestratorError(
+                    (f'Cannot automatically resolve ip address of host {host}. '
+                     + f'Please explicitly provide the address (ceph orch host add {host} --addr <ip-addr>)'))
```

Comment:
`if host in self.inventory` and `if old_ip != host` instead of the exception handling?